### DISCUSSION TITLE
Is there any reason that vsce package does not package any package.nls.json file?

### DISCUSSION THREAD

**Discussion by travis1111** [microsoft/vscode-discussions#1134] [Extension Development QnA] [answered]
I tried the l10n sample it works, package.nls.json is packaged in the vsix file.
But for some reason, the package.nls.json does not get packaged in my extension's vsix file.
What are the reasons?
I checked the json file is valid.

Thanks

**Comment by bwateratmsft** [comment, marked as answer]
Check your .vscodeignore file, to see if it is including the package.nls.json. The .vscodeignore file is like a .npmignore, it chooses what gets included in the VSIX bundle.

**Comment by alefragnani** [comment]
Hi @travis1111 ,

Since you played with the **l10n** sample, be sure you are looking the same localization resources, in the same place. I mean:

* `package.nls.xx.json`: These files are located in the root folder of your extension, in the same place as `package.json`, and contains localization for commands, settings and menus
* `l10n\bundle.l10n.xx.json`: These files are located within a folder (`l10n` on that sample) and contains localization for messages and other strings used inside your extension’s source

You can use `vsce ls` to see if your localization files are being recognized (or ignored by `.vscodeignore`, as @bwateratmsft mentioned), and their location. 

If the file is being packaged, but the localization is not working, be sure the strings used in the original `package.json` files are surrounded by `%`, like:  

```json
{   
     "title": "%separators.configuration.title%"
}
``` 

But has no `%` in `package.nls.json`, like:

```json
{
    "separators.configuration.title": "Separators"
}
```

Hope this helps
